lib/ronin/cli/pattern_options.rb

```diff
@@ -116,19 +116,67 @@ def self.included(command)
       #
       def self.define_numeric_options(command)
         command.option :number, short: '-N',
-                                desc: 'Searches for all numbers' do
-                                  @pattern = NUMBER
-                                end
+                                desc: 'Searches for all numbers',
+                                value: {
+                                  type: Regexp,
+                                  value: NUMBER
+                                }
```

Comment:
This breaks existing behavior. The `--number`/`-N` option should set `@pattern` to a Regexp.

All of the other options need a block which sets `@pattern` to the Regexp in the issue.